app/models/invenio_rdm_record_converter.rb

```diff
@@ -508,4 +515,19 @@ def funding(file_id)
     # remove nil values
     funding_sources.compact
   end
+
+
+  def map_collections_to_communities(communities=@communities)
+    mapped_communities = communities.flatten.map do |community|
+      @@collections_to_communites.map do |community_mapping_object|
+        if community[:id] == community_mapping_object["dh"]
+          community_mapping_object
+        else
+          nil
+        end
+      end
+    end
+
+    mapped_communities.flatten.compact
```

Comment:
For each collection path (first level array of `communities` here), there should be at most one Invenio community match. This means you can't flatten because you need to check each _path_ on it's own and not just the whole flattened set (whatever you want to call it).